src/goto-instrument/synthesizer/synthesizer_utils.h

```diff
@@ -50,7 +53,29 @@ get_loop_head(const unsigned int loop_number, goto_functiont &function);
 /// loops. Corresponding loops are specified by keys of `invariant_map`
 void annotate_invariants(
   const invariant_mapt &invariant_map,
-  goto_modelt &goto_model,
-  messaget &log);
+  goto_modelt &goto_model);
+
+/// Decide whether the target instruction is in the body of the transformed loop
+/// specified by `loop_id`.
+bool is_instruction_in_transfomed_loop(
+  const loop_idt &loop_id,
+  const goto_functiont &function,
+  unsigned location_number_of_target);
+
+/// Extract size of dynamic object from `expr`.
+std::size_t extract_size(const exprt &expr);
+
+/// Extract offset of dynamic object from `expr`.
+std::size_t extract_offset(const exprt &expr);
+
+/// convert hex string to size_t.
+std::size_t hex_to_size_t(const std::string &hex_str);
```

Comment:
I don't think you're using these anywhere? If you need them for some follow-up work: please move them to this follow-up work and then we'll discuss again (they need to be implemented differently, but I'd like to better understand how they are used so that I can give the most appropriate advice).